plugins/inputs/icinga2/icinga2.go

```diff
@@ -0,0 +1,158 @@
+package icinga2
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Icinga2 struct {
+	Server          string
+	ObjectType      string
+	Username        string
+	Password        string
+	ResponseTimeout internal.Duration
+	tls.ClientConfig
+
+	client *http.Client
+}
+
+type Result struct {
+	Results []Object `json:"results"`
+}
+
+type Object struct {
+	Attrs Attribute  `json:"attrs"`
+	Name  string     `json:"name"`
+	Joins struct{}   `json:"joins"`
+	Meta  struct{}   `json:"meta"`
+	Type  ObjectType `json:"type"`
+}
+
+type Attribute struct {
+	CheckCommand string `json:"check_command"`
+	DisplayName  string `json:"display_name"`
+	Name         string `json:"name"`
+	State        int    `json:"state"`
+}
+
+type ObjectType string
+
+var sampleConfig = `
+  ## Required Icinga2 server address (default: "https://localhost:5665")
+  # server = "https://localhost:5665"
+  
+  ## Required Icinga2 object type ("services" or "hosts, default "services")
+  # object_type = "services"
+
+  ## Credentials for basic HTTP authentication
+  # username = "admin"
+  # password = "admin"
+
+  ## Maximum time to receive response.
+  # response_timeout = "5s"
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = true
+  `
+
+func (i *Icinga2) Description() string {
+	return "Gather Icinga2 status"
+}
+
+func (i *Icinga2) SampleConfig() string {
+	return sampleConfig
+}
+
+func (i *Icinga2) GatherStatus(acc telegraf.Accumulator, checks []Object) {
+	for _, check := range checks {
+		fields := make(map[string]interface{})
+		tags := make(map[string]string)
+
+		fields["name"] = check.Attrs.Name
+		fields["state"] = check.Attrs.State
+
+		tags["display_name"] = check.Attrs.DisplayName
+		tags["check_command"] = check.Attrs.CheckCommand
+		tags["source"] = i.Server
+
+		acc.AddFields(fmt.Sprintf("icinga2_%s_status", i.ObjectType), fields, tags)
```

Comment:
Optional: consider dropping the `_status` suffix here, it feels superfluous.

Reply:
done